bool[] flag
---
int8 return
